t/ssl.t

```diff
@@ -2837,3 +2837,134 @@ lua ssl server name: "test.com"
 [error]
 [alert]
 [emerg]
+
+=== TEST 29: parse PEM cert and key to cdata
+--- http_config
+    lua_package_path "$TEST_NGINX_LUA_PACKAGE_PATH";
+
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/nginx.sock ssl;
+        server_name   test.com;
+        ssl_certificate_by_lua_block {
+            local ssl = require "ngx.ssl"
+
+            ssl.clear_certs()
+
+            local f = assert(io.open("t/cert/chain/chain.der"))
+            local cert_data = f:read("*a")
+            f:close()
+
+            local cert, err = ssl.parse_der_cert(cert_data)
+            if not cert then
+                ngx.log(ngx.ERR, "failed to parse pem cert: ", err)
+                return
+            end
+
+            local ok, err = ssl.set_cert(cert)
+            if not ok then
+                ngx.log(ngx.ERR, "failed to set cert: ", err)
+                return
+            end
+
+            local f = assert(io.open("t/cert/chain/test-com.key.der"))
```

Comment:
did you forgot t/cert/chain/test-com.key.der

Reply:
No - this file has existed for years https://github.com/openresty/lua-resty-core/blob/master/t/cert/chain/test-com.key.der